### PR TITLE
swap: refactor message handlers to be on

### PR DESCRIPTION
**Pull request by holisticode**
This PR addresses a change request by @acud in PR https://github.com/ethersphere/swarm/pull/1554/.

It moves message handlers from `Peer` to `Swap`  by taking `network/retrieval` as template, as requested.

Adapted all tests as well.

### CODE REVIEW

**Comment by holisticode** [comment]
Closes https://github.com/ethersphere/swarm/issues/1646

File: swap/swap.go [hunk 1]
```diff
@@ -179,6 +179,146 @@ func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
...
+	if err != nil {
+		return err
+	}
+
+	log.Debug("received cheque processed and verified", "peer", p.ID().String())
+
+	// reset balance by amount
+	// as this is done by the creditor, receiving the cheque, the amount should be negative,
+	// so that updateBalance will calculate balance + amount which result in reducing the peer's balance
+	err = s.resetBalance(p.ID(), 0-int64(cheque.Honey))
```
Comment (ralph-pichler) [inline swap/swap.go:209]: This was not introduced in this PR but re-reading this we should probably lock s here (or inside `resetBalance`).

Comment (holisticode) [reply swap/swap.go:209]: Great catch man, I agree. Will post an update

**Review by mortelli — APPROVED** [review]
LGTM: the changes in function signatures seem to match what was done in `network/retrieval`, and all the tests are passing.

Before merging, I think we should probably do a manual local simulation to make sure this doesn't break anything (just to be safe).

File: swap/protocol_test.go [hunk 2]
```diff
@@ -236,22 +237,26 @@ func TestTriggerPaymentThreshold(t *testing.T) {
 	overDraft := 42
 	debitorSwap.balances[creditor.ID()] = 0 - DefaultPaymentThreshold
 
+	fmt.Println("1")
```
Comment (acud) [inline swap/protocol_test.go:240]: @holisticode please remove these on the main incentives pr.

**Pull request merged**